internal/controllers/common/resource_definitions/resource_definitions.go

```diff
@@ -976,6 +981,27 @@ func NewConsoleLink(cr *operatorv1beta1.Cryostat, url string) *consolev1.Console
 	}
 }
 
+func SetAdditionalCorsAllowedOrigins(specs *ServiceSpecs) (*configv1.APIServer, error) {
```

Comment:
As of now, I can deploy the operator without any errors, but I don't see an `AdditionalCORSAllowedOrigins` spec when I run `oc describe apiserver cluster`. I also don't see any errors in the `cryostat-operator-controller-manager` logs. 

Would greatly appreciate any tips to debug this, thanks!
```
$ oc describe apiserver cluster
Name:         cluster
Namespace:    
Labels:       <none>
Annotations:  include.release.openshift.io/ibm-cloud-managed: true
              include.release.openshift.io/self-managed-high-availability: true
              include.release.openshift.io/single-node-developer: true
              oauth-apiserver.openshift.io/secure-token-storage: true
              release.openshift.io/create-only: true
API Version:  config.openshift.io/v1
Kind:         APIServer
Metadata:
  Creation Timestamp:  2021-12-17T21:24:58Z
  Generation:          5
  Managed Fields:
    API Version:  config.openshift.io/v1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .:
          f:include.release.openshift.io/ibm-cloud-managed:
          f:include.release.openshift.io/self-managed-high-availability:
          f:include.release.openshift.io/single-node-developer:
          f:oauth-apiserver.openshift.io/secure-token-storage:
          f:release.openshift.io/create-only:
      f:spec:
        .:
        f:audit:
          .:
          f:profile:
    Manager:         cluster-version-operator
    Operation:       Update
    Time:            2021-12-17T21:24:58Z
  Resource Version:  2202263
  UID:               a6c97a3c-d651-48a8-80b0-3d10cfb58ef3
Spec:
  Audit:
    Profile:  Default
```

Reply:
Here are some of the other pieces of code that I've tried using the existing `client.Client` in the reconciler. These snippets all compile, don't output any errors in the cryostat operator logs, and also don't update the APIServer spec:
```
// cryostat_controller.go, around line 356
if r.IsOpenShift {
... 
 apiServerConfig := &configv1.APIServer{
            ObjectMeta: metav1.ObjectMeta{
                Name: "cluster",
            },
            Spec: configv1.APIServerSpec{
                AdditionalCORSAllowedOrigins: []string{serviceSpecs.CoreURL.Hostname()},
            },
        }

        _, err = controllerutil.CreateOrUpdate(ctx, r.Client, apiServerConfig, func() error {
            apiServerConfig.Spec = configv1.APIServerSpec{
                AdditionalCORSAllowedOrigins: []string{serviceSpecs.CoreURL.Hostname()},
            }
            return nil
        })
        if err != nil {
            return reconcile.Result{}, err
        }
...
```
```
// cryostat_controller.go, around line 356
if r.IsOpenShift {
... 
     apiServer := &configv1.APIServer{}
     err = r.Client.Get(context.Background(), types.NamespacedName{Name: "cluster"}, apiServer)
     if err != nil && errors.IsNotFound(err) {
         return reconcile.Result{}, err
     }

     apiServerSpecPatch := resources.NewCorsOriginForApiServer(serviceSpecs)
     err = r.Client.Status().Update(context.Background(), apiServer)
     if err != nil {
         return reconcile.Result{}, err
    }
...

// in resource_definitions.go
func NewCorsOriginForApiServer(specs *ServiceSpecs) client.Patch {

     // Allow access to the OAuth server using CORS headers

     apiServerData := &configv1.APIServer{
         ObjectMeta: metav1.ObjectMeta{
             Name: "cluster",
         },
         Spec: configv1.APIServerSpec{
             AdditionalCORSAllowedOrigins: []string{specs.CoreURL.Hostname()},
        },
     }

     return client.MergeFrom(apiServerData.DeepCopy())
 }
```